sender/batch.go

```diff
@@ -60,19 +60,29 @@ This means that:
 3. Send() will only block if two channels are full.
 */
 type Batch struct {
-	Next      skogul.SenderRef `doc:"Sender that will receive batched metrics"`
-	Interval  skogul.Duration  `doc:"Flush the bucket after this duration regardless of how full it is"`
-	Threshold int              `doc:"Flush the bucket after reaching this amount of metrics"`
-	allocSize int
-	ch        chan *skogul.Container
-	once      sync.Once
-	metrics   int
-	timer     *time.Timer
-	cont      *skogul.Container
-	out       chan *skogul.Container
+	Next       skogul.SenderRef `doc:"Sender that will receive batched metrics"`
+	Interval   skogul.Duration  `doc:"Flush the bucket after this duration regardless of how full it is"`
+	Threshold  int              `doc:"Flush the bucket after reaching this amount of metrics"`
+	allocSize  int
+	ch         chan *skogul.Container
+	once       sync.Once
+	metrics    int
+	timer      *time.Timer
+	cont       *skogul.Container
+	out        chan *skogul.Container
+	numThreads int
```

Comment:
This can be public/exported, having a capital letter. The `doc` field should also be filled in, as for "Next", "Interval" and "Threshold", so documentation is created for it.

Reply:
I guess you're suggesting this change:

```
NumThreads int `doc:"Number of threads for batch sender"`
```

Is this what you want it to be, or do you need any other modifications?

Reply:
This looks good! To be in line with other configuration options regarding threads, we should name it "Threads" instead of "NumThreads", such as in the UDP receiver: https://github.com/telenornms/skogul/blob/primary/receiver/udp.go#L48

Reply:
You could also add docs about the default (default: uses the number of CPU cores). :)

Reply:
Ah, didn't notice this, do you want me to fix the docs? I'll add this in the next commit, if you wish.

Reply:
Yes please :)

Reply:
Done! 😄